alica_engine/include/engine/AlicaContext.h

```diff
@@ -411,6 +411,19 @@ class AlicaContext
     template <class T>
     bool setOptions(const std::vector<std::pair<std::string, T>>& keyValuePairs, bool reload = true) noexcept;
 
+    /**
+     * Set logger to be used by this alica framework instance.
+     * Example usage: setLogger<alica::AlicaDefaultLogger>();
+     *
+     * @note LoggerType must be a derived class of IAlicaLogger
+     * @note This must be called before initializing context
+     *
+     * @param args Additional arguments to be forwarded to constructor of logger. Might be empty.
+     * The first two arguments, verbosity and agentName, will always be passed to the constructor of the logger.
+     */
+    template <class LoggerType, class... Args>
+    void setLogger(Args&&... args);
```

Comment:
I think this is tricky, because we use the member variables _localAgentName, which can be passed to the static setLogger function, and _configRootNode. For the config, we would have to pass the required information to parse the config to setLogger, parse the config to then read the verbosity level. Then we have to parse the config again in the initializer list of the AlicaContext.

Reply:
you could take the alica context as a parameter, but that may make the API a bit confusing, so lets leave it as is, but mention in a comment that logger will be static & limited to 1 instance per process